dom.d.ts

```diff
@@ -0,0 +1,44 @@
+type Nil = null | undefined;
+type Primitive = string | number | bigint | boolean;
+
+type Prop = Nil | Primitive;
+type Props = Nil | Record<string, Prop>;
```

Comment:
A more advanced version of `Props` could also include "well-known" special cases, which can be found in the private function `setProp`. For example, `dataset` can be only `Record<string, Stringable>`.